### ISSUE TITLE
Mac OS X - process.WaitForExit() does not work.

### ISSUE THREAD

**Issue by 616E64726173** [dotnet/runtime#32456]
Version: DotNet Core 3.1
OS: MacOSX Catalina

Right now I am using "open -W" as a workaround but it's not ideal. 
Additionally, differences in how MacOS handles Applications means that even if it did work as intended, in most cases we actually want to monitor the spawned child process. Opening TextEdit will in fact invoke TextEdit, but will wait on the entire TextEdit application to exit and not just the opened file. 

Correct me if I am wrong, but there is presently no way in .NET to obtain the child PID except by calling Process.GetProcesses() and hoping you select the right one.

Expected behavior: For it to work.
Actual behavior: It does not work.
Ideal behavior: Return the child process if there is one, likely not possible since the invoked application will be the only one aware of the PID.

```
            using (var process = Process.Start("chrome", "www.bing.com"))
            {
                process.WaitForExit();
            }
```
Some Googling reveals people encountering similar issues.

[https://stackoverflow.com/questions/55639285/process-waitforexit-is-not-waiting-in-mac-os-x-asp-net-core-2-2](https://stackoverflow.com/questions/55639285/process-waitforexit-is-not-waiting-in-mac-os-x-asp-net-core-2-2)

[dotnet/corefx/Process.OSX.cs:48](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs#L48) shows that it is invoking open; so I am getting the PID of open and not even the spawned application, explaining why Exited immediately fires.

**Comment by scalablecory** [comment]
@wfurt

**Comment by wfurt** [comment]
> TextEdit will in fact invoke TextEdit, but will wait on the entire TextEdit application to exit and not just the opened file.

That is how the Process API work. You can know when a process is finished but it does not give out information when a particular action is taken or completed.  It seems like the difference is that Linux would start fresh process but on OSX it could reuse an existing app.
So as there is a difference if the app exits when done processing a given task or not. 
It seems like you would need to find a way how to tell the app to exit. 

If you feel there is an issue with `WaitForExit`, please post runnable repro.

**Comment by 616E64726173** [comment]
Yes, there is an issue with WaitForExit on OS X. It doesn't work at all.

As I said, in [dotnet/corefx/Process.OSX.cs:48 ](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs#L48)you can clearly see that it invokes /usr/bin/open to spin up a process.

[dotnet/corefx/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:1333](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L1333) simply waits for the exit event to be raised, which happens almost immediately as soon as /usr/bin/open completes launching the child process because it's monitoring the PID and events for 'open' and not the process it launches.

**Comment by amsoedal** [comment]
I'm also interested in a fix for this! We were able to repro this using `bool WaitForExit(int milliseconds)`, but when we switched to `void WaitForExit()` it seemed to work.
